### PR TITLE
Feature/ab11519 autocomplete

### PR DESCRIPTION
**Pull request by CarloBar1**
Feature [AB#11519](https://dev.azure.com/neovici/7e94365d-32b1-416f-854b-7f195da31da6/_workitems/edit/11519) - new contour styling applied to the chip in cosmoz-autocomplete.

https://github.com/Neovici/cosmoz-autocomplete/assets/122988480/8927e7b9-799b-4653-8135-9312717f876c

### CODE REVIEW

File: stories/cosmoz-autocomplete.stories.js [hunk 1]
```diff
@@ -104,3 +104,29 @@ export const wrap = () => html` ${css}
 		wrap
 		style="max-width: 170px"
 	></cosmoz-autocomplete>`;
+
+export const contour = () => html`<style>
+		@import url('https://fonts.googleapis.com/css2?family=Inter:wght@100;300;400;500&display=swap');
+		cosmoz-autocomplete,
+		cosmoz-listbox {
+			font-family: 'Inter', sans-serif;
+		}
```
Comment (cristinecula) [inline stories/cosmoz-autocomplete.stories.js:113]: Why did you include a font here?

File: src/autocomplete/chip.css.ts [hunk 2]
```diff
@@ -13,6 +13,7 @@ export default css`
 		min-width: 40px;
 		padding: 0 4px 0 8px;
 		gap: 4px;
+		transform: translateY(var(--cosmoz-autocomplete-chip-translate-y));
```
Comment (cristinecula) [inline src/autocomplete/chip.css.ts:16]:
```suggestion
		transform: translateY(var(--cosmoz-autocomplete-chip-translate-y, 0));
```

**Review by cristinecula — CHANGES_REQUESTED** [review]

File: stories/cosmoz-autocomplete.stories.js (hunk 1 above)
Comment (CarloBar1) [reply stories/cosmoz-autocomplete.stories.js:113]: It is a style spec from the FIgma mockups and UX meetings, and I noticed some issues in fixing this font from frontend.

File: src/autocomplete/chip.css.ts (hunk 2 above)
Comment (CarloBar1) [reply src/autocomplete/chip.css.ts]: Addressed

File: stories/cosmoz-autocomplete.stories.js (hunk 1 above)
Comment (cristinecula) [reply stories/cosmoz-autocomplete.stories.js:113]: But there is already the `css` that the other stories use. Why don't you use that one?

Comment (CarloBar1) [reply stories/cosmoz-autocomplete.stories.js:127]: There seems to be something wrong with the `@import` rule there. Anyways I moved the font definition there, the font import I need is performed in Frontend.

File: src/autocomplete/chip.css.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@ export default css`
 		min-width: 40px;
 		padding: 0 4px 0 8px;
 		gap: 4px;
-		cursor: pointer;
```
Comment (cristinecula) [inline src/autocomplete/chip.css.ts:16]: You dropped `cursor: pointer;` during rebase, please put it back.

File: src/autocomplete/styles.css.ts [hunk 4]
```diff
@@ -5,6 +5,7 @@ export default css`
 		display: block;
 		position: relative;
 		min-width: 35px;
+		font-family: 'Inter', sans-serif;
```
Comment (cristinecula) [inline src/autocomplete/styles.css.ts:8]: If you do this, you are breaking the cascade, forcing authors to either load the Inter font OR write `cosmoz-autocomplete { font-family: myfont-family, etc }`. Revert this.

File: stories/cosmoz-autocomplete.stories.js (hunk 1 above)
Comment (cristinecula) [reply stories/cosmoz-autocomplete.stories.js:126]: You are duplicating the same code that is in the `css` variable at the top of this file.
https://github.com/Neovici/cosmoz-autocomplete/blob/ab57478cef04c6df23834b148ec7a7252461ce1d/stories/cosmoz-autocomplete.stories.js#L9-L16

Why aren't you just interpolating that value, just like in the other stories? Why do you need to duplicate it?

Pay attention to the fact that it is already wrapped in a `<style>` tag. You might have tried to do something like `<style>${css}</style>`, in which case it would have resulted in `<style><style>...</style></style>`, which of course won't work properly.

**Review by cristinecula — CHANGES_REQUESTED** [review]

File: stories/cosmoz-autocomplete.stories.js [hunk 5]
```diff
@@ -107,16 +107,39 @@ export const wrap = () =>
...
+		cosmoz-autocomplete {
+			--cosmoz-input-color: #aeacac;
+			--cosmoz-input-border-radius: 4px;
+			--cosmoz-input-padding: 12px;
+			--cosmoz-input-line-display: none;
+			--cosmoz-input-contour-size: 1px;
+			--cosmoz-input-label-translate-y: 35%;
```
Comment (cristinecula) [inline stories/cosmoz-autocomplete.stories.js:133]: Make it consistent with the cosmoz-input stories:

```suggestion
			--cosmoz-input-label-translate-y: 10px;
```

**Review by cristinecula — CHANGES_REQUESTED** [review]

File: src/autocomplete/styles.css.ts (hunk 4 above)
Comment (CarloBar1) [reply src/autocomplete/styles.css.ts:8]: Addressed

File: stories/cosmoz-autocomplete.stories.js (hunk 5 above)
Comment (CarloBar1) [reply stories/cosmoz-autocomplete.stories.js:133]: Addressed

File: stories/cosmoz-autocomplete.stories.js (hunk 1 above)
Comment (CarloBar1) [reply stories/cosmoz-autocomplete.stories.js:126]: OK, consistently with the other comment in the style file, it's probably best if this font styling is set directly in frontend, via e.g. vars - I recall you mentioning the latter being more effective, I will add a var for it so it's accessible from the outside.

Comment (CarloBar1) [reply stories/cosmoz-autocomplete.stories.js:126]: Or, to avoid adding more code, already existing parts might be a better solution. I recall you mentioning that vars are more effective than parts, but they're already used and working in frontend.

File: src/autocomplete/chip.css.ts (hunk 3 above)
Comment (CarloBar1) [reply src/autocomplete/chip.css.ts:16]: Addressed

File: stories/cosmoz-autocomplete.stories.js [hunk 6]
```diff
@@ -107,16 +107,34 @@ export const wrap = () =>
...
 		.label=${'Choose color'}
 		.source=${colors}
 		.limit=${1}
 		.textProperty=${'text'}
-		.value=${colors[2]}
-		show-single
-		preserve-order
-		.min=${1}
-	></cosmoz-autocomplete>
-`;
```
Comment (cristinecula) [inline stories/cosmoz-autocomplete.stories.js:122]: Please revert the changes to the `select` story.

File: stories/cosmoz-autocomplete.stories.js [hunk 7]
```diff
@@ -107,16 +107,34 @@ export const wrap = () =>
...
+		.value=${[colors[0], colors[1], colors[2]]}
+		wrap
+		style="max-width: 170px"
+	></cosmoz-autocomplete>`;
+
+export const contour = () => html`<style>
```
Comment (cristinecula) [inline stories/cosmoz-autocomplete.stories.js:121]: Add the font style css. It will work after upgrading cosmoz-input (https://neovici.slack.com/archives/C6LJQMJFM/p1711093197592089)

```suggestion
export const contour = () => html`${css}<style>
```

**Review by cristinecula — CHANGES_REQUESTED** [review]

File: stories/cosmoz-autocomplete.stories.js (hunk 7 above)
Comment (CarloBar1) [reply stories/cosmoz-autocomplete.stories.js]: Addressed, the font works as expected!

**Review by cristinecula — APPROVED** [review]
Very good!
